### ISSUE TITLE
Kiev/Kyiv - Invalid time zone definition for start time. [400=BadRequest]

### ISSUE THREAD

**Issue by StanislavCha** [phw198/OutlookGoogleCalendarSync#1525]
OGCS Version: v 2.9.5.0
Edition: Installed 
Sync Direction: O->G

**Bug Description**  
Starting from a couple of days ago, all new events do not sync and provide the following error: Invalid time zone definition for start time. [400=BadRequest]
[](url
[OGcalsync.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/9336143/OGcalsync.log)
)

**Comment by MaxWellBoss** [comment]
Same problem
[OGcalsync.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/9336646/OGcalsync.log)

**Comment by pkaristo** [comment]
Same issue. For me it seems to occur due to changed time zone defintion. The difference I notice is whether the city list includes Kiev or Kyiv. Did google change their spelling choice and hence it does not work?

Edit: Indeed, started to happen last week.

**Comment by henrivirmo** [comment]
> Same issue. For me it seems to occur due to changed time zone defintion. The difference I notice is whether the city list includes Kiev or Kyiv. Did google change their spelling choice and hence it does not work?

Same issue started happening last week. Also in the same time zone as pkaristo.

**Comment by mawid6** [comment]
# Showstopper! 

Same problem. Syncs no longer complete due to this error. Also Kiev timezone in the error.

[OGcalsync.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/9349237/OGcalsync.log)
 
<img width="383" alt="image" src="https://user-images.githubusercontent.com/77722885/184839345-04a30ddc-7221-45c1-890f-e23b0cbbb04e.png">

**Comment by KertuMikk** [comment]
same issue

**Comment by andreska** [comment]
Same issue

**Comment by krisavi** [comment]
I have same issue. For new calendar items it seems to throw error
```
New event failed to save.
Invalid time zone definition for start time. [400=BadRequest]
```

On existing ones, it shows the small table with Attribute of Start and End Timezone being changed from Europe/Kiev to Europe/Kyiv. From that it seems Google has changed the name of the time zone and it is now not working.

https://stackoverflow.com/questions/71829570/how-and-where-should-i-suggest-a-rename-of-an-existing-timezone-name could be related if Google takes their information from IANA TZ DB and MS Outlook not getting update so soon to it.

**Comment by phw198** [comment]
Looks like IANA changed their definition on 15-Aug - their release notes for August say:

>  Changes to zone name: Rename Europe/Kiev to Europe/Kyiv, as "Kyiv" is more common in English now.  Spelling of other names in Ukraine has not yet demonstrably changed in common English practice so for now these names retain old spellings, as in other countries (e.g., Europe/Prague not "Praha", and Europe/Sofia not "Sofiya"). 

I guess Google still has the old definition of 'Kiev' then, so probably best to get the version of tzdb.dat out of the portable version and replace the recently updated version within your installs. 

1. Download [tzdb-2022a.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/9439938/tzdb-2022a.zip)
2. Extract the tzdb.nzd file and replace the file in your OGCS installation directory
3. Right click the extracted file, Properties and set it as read-only (this will stop it being updated to the latest version by OGCS)

**Comment by maremae** [comment]
> I guess Google still has the old definition of 'Kiev' then, so probably best to get the version of tzdb.dat out of the portable version and replace the recently updated version within your installs.

Indeed, I had luck replacing the tzdb.nzd file at \AppData\Local\OutlookGoogleCalendarSync\app-2.9.5-alpha with the one (version 2022a, I believe) acquired from the portable variant. Only thing is that I had to set the file as read-only for the time being, since it was being overwritten with the 2022c on OGCS startup.

**Comment by MONAtLEWIS** [comment]
Same Issue

**Comment by sysinoro** [comment]
Same issue here. Is there going to be a fix for this?

**Comment by pkaristo** [comment]
I managed to fix the issue, at least for now, with the guidelines given by @maremae above.
-I closed the OGCS application.
-Then I downloaded the portable of the latest alpha release (https://github.com/phw198/OutlookGoogleCalendarSync/releases/download/v2.9.5-alpha/Portable_OGCS_v2.9.5.zip)
- unpacked the zipped folder
- then copied the file from there (\Portable_OGCS_v2.9.5\tzdb.nzd) to the install folder of the application in my pc (\AppData\Local\OutlookGoogleCalendarSync\app-2.9.0-beta) replacing the similarly named file
- I also set the new file as read-only, just in case.
- Then restarted OGCS and it synced without issues

I noticed I lost at least one of the calendar entries that were conflicting.

**Comment by MONAtLEWIS** [comment]
Yes. It is definitely the tzdb.nzd file that needs to be updated in the setup.exe installer.

**Comment by artisticcheese** [comment]
> * \app-2.9.0-beta) replacing the similarly named file

This did not work for me, still complaining

**Comment by mterho** [comment]
> 

I also replaced the tzdb.nzd file. Did not work. Still complaining.

**Comment by pkaristo** [comment]
Hmm, interesting that  it did not fix it for everyone. Could it be because I got a double fix? Ok, that's a partial joke, but there is something strange going on. I only did the file replacement on my work PC which then started to work correctly, BUT I also did not get the error any more on my  home PC after that, evethough I didn't replace any files here. I'm using the same calendar accounts in both PC's, so is it possible that there was some kind of mismatch between the versions of my two PCs? Otherwise it would be strange that the error disappeared from both computers.

**Comment by MateFarkas13** [comment]
> I managed to fix the issue, at least for now, with the guidelines given by @maremae above. -I closed the OGCS application. -Then I downloaded the portable of the latest alpha release (https://github.com/phw198/OutlookGoogleCalendarSync/releases/download/v2.9.5-alpha/Portable_OGCS_v2.9.5.zip)
> 
> * unpacked the zipped folder
> * then copied the file from there (\Portable_OGCS_v2.9.5\tzdb.nzd) to the install folder of the application in my pc (\AppData\Local\OutlookGoogleCalendarSync\app-2.9.0-beta) replacing the similarly named file
> * I also set the new file as read-only, just in case.
> * Then restarted OGCS and it synced without issues
> 
> I noticed I lost at least one of the calendar entries that were conflicting.

Worked for me 👍

**Comment by vatioz** [comment]
Same issue here, also started happening recently,
but **I didn't find** any mention of **Kyiv nor Kiev** in my OGCS logfile. Here is excerpt from my log:

```
2022-08-24 09:08:36,353 FAIL  22 OutlookGoogleCalendarSync.GoogleOgcs.Calendar [1764] -  Google.Apis.Requests.RequestError
Invalid time zone definition for start time. [400]
Errors [
	Message[Invalid time zone definition for start time.] Location[ - ] Reason[invalid] Domain[global]
	Message[Invalid time zone definition for end time.] Location[ - ] Reason[invalid] Domain[global]
]

2022-08-24 09:08:36,356 WARN  22 OutlookGoogleCalendarSync.GoogleOgcs.Calendar [1820] -  Unhandled API exception.
2022-08-24 09:08:36,359 ERROR  22 OutlookGoogleCalendarSync.Console [278] -  New event failed to save.
2022-08-24 09:08:36,381 ERROR  22 OutlookGoogleCalendarSync.Console [278] -  Invalid time zone definition for start time. [400=BadRequest]
```

**Comment by artisticcheese** [comment]
I guess the reason is a war in Ukraine and change of transliteration from Russian sounding names to Ukranian sounding one https://clickup.canny.io/feature-requests/p/please-rename-europekiev-timezone-to-europekyiv
This shall break so many things though

**Comment by Routout** [comment]
> I managed to fix the issue, at least for now, with the guidelines given by @maremae above. -I closed the OGCS application. -Then I downloaded the portable of the latest alpha release (https://github.com/phw198/OutlookGoogleCalendarSync/releases/download/v2.9.5-alpha/Portable_OGCS_v2.9.5.zip)
> 
>     * unpacked the zipped folder
> 
>     * then copied the file from there (\Portable_OGCS_v2.9.5\tzdb.nzd) to the install folder of the application in my pc (\AppData\Local\OutlookGoogleCalendarSync\app-2.9.0-beta) replacing the similarly named file
> 
>     * I also set the new file as read-only, just in case.
> 
>     * Then restarted OGCS and it synced without issues
> 
> 
> I noticed I lost at least one of the calendar entries that were conflicting.

This worked for me after I changed tzdb.nzd to read only. So it seem to be necessary.

**Comment by artisticcheese** [comment]
> 

Confirming. Setting file to read-only was in fact required even though does not make any sense since file itself does not seem to be mofied

**Comment by Routout** [comment]
> Confirming. Setting file to read-only was in fact required even though does not make any sense since file itself does not seem to be mofied

Newer 2.9.5 file is 133kb, but if its not read only, after restarting software it's 129kb.

**Comment by mterho** [comment]
Confirming. The read-only attribute for tzdb.nzd fixed the issue. Thank you All!

**Comment by phw198** [comment]
:package: This should now be resolved with [hotfix v2.9.5.11.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/9440012/v2.9.5.11.zip) ([see instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix)).

If you have applied the workaround of reverting the `tzdb.nzd` and setting it read only, you would need to set it back to read-write to check the hotfix is actually working.

Let me know how it goes!

**Comment by Routout** [comment]
> 📦 This should now be resolved with [hotfix v2.9.5.11.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/9440012/v2.9.5.11.zip) ([see instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix)).
> 
> If you have applied the workaround of reverting the `tzdb.nzd` and setting it read only, you would need to set it back to read-write to check the hotfix is actually working.
> 
> Let me know how it goes!

Thank you. 2.9.5.11 seem to be working for me.

**Comment by henrivirmo** [comment]
Thanks for the hotfix! Managed to get it working after some wrestling with the antivirus software. Sync went perfectly even after such a long time! 😁
